code/modules/plumbing/shower.dm

```diff
@@ -0,0 +1,223 @@
+	icon_state = "shower"
+	density = 0
+	anchored = 1
+	use_power = 0
+	var/on = 0
+	var/obj/effect/mist/mymist = null
+	var/ismist = 0				//needs a var so we can make it linger~
+	var/watertemp = "normal"	//freezing, normal, or boiling
+
+/obj/machinery/shower/Initialize()
+	create_reagents(5)
+	..()
+
+
+/obj/effect/mist
+	name = "mist"
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "mist"
+	layer = FLY_LAYER
+	anchored = 1
+	mouse_opacity = 0
+
+
+/obj/machinery/shower/attack_hand(mob/M)
+	if(!on && !plumbing_has_reagents(5) && !reagents.total_volume)
+		M << "<span class='warning'>Nothing comes out of the showerhead!</span>"
+		return
+	on = !on
+	update_icon()
+	add_fingerprint(M)
+	master_plumber.request_liquid(src, 5)
+	if(on)
+		process()
+
+
+/obj/machinery/shower/attackby(obj/item/I, mob/user, params)
+	if(I.type == /obj/item/device/analyzer)
+		to_chat(user, "<span class='notice'>The water temperature seems to be [watertemp].</span>")
+	if(istype(I, /obj/item/weapon/wrench))
+		to_chat(user, "<span class='notice'>You begin to adjust the temperature valve with \the [I]...</span>")
+		if(do_after(user, 50*I.toolspeed, target = src))
+			switch(watertemp)
+				if("normal")
+					watertemp = "freezing"
+				if("freezing")
+					watertemp = "boiling"
+				if("boiling")
+					watertemp = "normal"
+			user.visible_message("<span class='notice'>[user] adjusts the shower with \the [I].</span>", "<span class='notice'>You adjust the shower with \the [I] to [watertemp] temperature.</span>")
+			log_game("[key_name(user)] has wrenched a shower to [watertemp] at ([x],[y],[z])")
+			add_hiddenprint(user)
+
+
+/obj/machinery/shower/update_icon()	//this is terribly unreadable, but basically it makes the shower mist up
+	cut_overlays()					//once it's been on for a while, in addition to handling the water overlay.
+	if(mymist)
+		qdel(mymist)
+
+	if(on)
+		add_overlay(image('icons/obj/watercloset.dmi', src, "water", MOB_LAYER + 1, dir))
+		if(watertemp == "freezing")
+			return
+		if(!ismist)
+			spawn(50)
+				if(src && on)
+					ismist = 1
+					mymist = new /obj/effect/mist(loc)
+		else
+			ismist = 1
+			mymist = new /obj/effect/mist(loc)
+	else if(ismist)
+		ismist = 1
+		mymist = new /obj/effect/mist(loc)
+		spawn(250)
+			if(!on && mymist)
+				qdel(mymist)
+				ismist = 0
+
+
+/obj/machinery/shower/Crossed(atom/movable/O)
+	..()
+	if(on)
+		if(isliving(O))
+			var/mob/living/L = O
+			if(wash_mob(L)) //it's a carbon mob.
+				var/mob/living/carbon/C = L
+				C.slip(4,2,null,NO_SLIP_WHEN_WALKING)
+		else
+			wash_obj(O)
+
+
+/obj/machinery/shower/proc/wash_obj(atom/movable/O)
+	O.clean_blood()
+
+	if(istype(O,/obj/item))
+		var/obj/item/I = O
+		I.acid_level = 0
+		I.extinguish()
+
+
+/obj/machinery/shower/proc/wash_turf()
+	if(isturf(loc))
+		var/turf/tile = loc
+		loc.clean_blood()
+		for(var/obj/effect/E in tile)
+			if(is_cleanable(E))
+				qdel(E)
+
+
+/obj/machinery/shower/proc/wash_mob(mob/living/L)
+	L.wash_cream()
+	if(iscarbon(L))
```

Comment:
I cut the fire stack removal out of here, as I think it's what water does anyhow.